shell/platform/darwin/ios/framework/Headers/FlutterEngine.h

```diff
@@ -58,7 +58,7 @@ FLUTTER_EXPORT
  *   the threads used by this FlutterEngine.
  * @param projectOrNil The `FlutterDartProject` to run.
  */
-- (instancetype)initWithName:(NSString*)labelPrefix project:(FlutterDartProject*)projectOrNil;
+- (nonnull instancetype)initWithName:(nonnull NSString*)labelPrefix project:(nullable FlutterDartProject*)projectOrNil;
```

Comment:
This can return null

Reply:
you mean the return type? Can it?

Reply:
It _shouldn't_, but the impl has `if (self)` in there. Maybe someone more familiar with ObjC and Swift could help on this - my inclination is that becuase the impl explicitly assumes that `[super init]` can return null, we should treat it as such, but I found something in the docs suggesting that `NSObject init` is never expected to return null.  @jmagman might know.

Reply:
`-[NSObject init]` is actually not decorated with nonnull but should be assumed to be nonnull.  So unless this initializer actually returns null in some cases (common enough, see `-[NSCoding initWithCoder:]`) or it's baseclass designated initializer can return null, you can assume nonnull.

Reply:
I think it's a convention but since we know this is NSObject, I don't think it can return nil

https://opensource.apple.com/source/objc4/objc4-532/runtime/NSObject.mm (it just returns self). Other supers can return nil but we also NSAssert non-nil (I don't think we have DNS_BLOCK_ASSERTIONS turned on anywhere)

Reply:
In that case, should we just change the impl to assert not nil, and remove the if self as part of this patch?

Reply:
it already asserts non nil and doesn't if self (if I understood you correctly)

Reply:
Ahh right.  Ok, I was misremembering.